modules/standard/DynamicIters.chpl

```diff
@@ -699,12 +699,15 @@ where tag == iterKind.follower
 //************************* Helper functions
 private proc defaultNumTasks(nTasks:int)
 {
-  var dnTasks=nTasks;
-  if nTasks==0 then {
-    if dataParTasksPerLocale==0 then dnTasks=here.maxTaskPar;
-      else dnTasks=dataParTasksPerLocale;
-  } else if nTasks<0 then {
-    halt("'numTasks' is negative");
+  var dnTasks = nTasks;
+  if nTasks <= 0  {
+    if dataParTasksPerLocale == 0 then
+      dnTasks = here.maxTaskPar;
+    else
+      dnTasks = dataParTasksPerLocale;
+
+    if nTasks < 0 then
+      warning("'numTasks' < 0, defaulting to '" + dnTasks + "' tasks");
```

Comment:
Did you really want the value of dnTasks to appear in single-quotes in the message?  For example,

```
'numTasks' < 0, defaulting to '8' tasks
```